addons/addon-raas-s3-copy/packages/s3-synchronizer/src/s3-download.go

```diff
@@ -124,34 +282,49 @@ func getObjectsAll(
 			continue
 		}
 
-		if debug == true {
-			log.Printf("%v -> %v\n", *item.Key, destFilePath)
-		}
-
 		// Ensure the directory exists
 		os.MkdirAll(filepath.Dir(destFilePath), 0775)
 
-		downloader := s3manager.NewDownloader(sess, func(d *s3manager.Downloader) {
-			d.PartSize = 100 * 1024 * 1024
-			d.Concurrency = concurrency
-		})
+		shouldDownload := true
+
+		// Note that we cannot use os.IsExist(fileError) to check for file's existence
+		// os.IsExist and os.IsNotExist are error checker functions and only work when error is not nil
+		// The correct way to check if file exists is using !os.IsNotExist(fileError)
+		if _, fileError := os.Stat(destFilePath); !os.IsNotExist(fileError) {
+			// If the file has not changed in S3 since last download then skip downloading it
+			shouldDownload = hasFileChangedInS3(item)
+			if !shouldDownload && debug {
+				log.Printf("'%v' already exists and is up-to-date. Skip downloading '%v'\n", destFilePath, *item.Key)
+			}
+		}
+		if !shouldDownload {
+			continue
+		}
+
+		if debug {
+			log.Printf("%v -> %v\n", *item.Key, destFilePath)
+		}
 
 		destFile, err := os.Create(destFilePath)
 		if err != nil {
-			if debug == true {
+			if debug {
 				log.Println("Create file error: ", err.Error())
 			}
 			stats.errorPrefixes = append(stats.errorPrefixes, item.Key)
 			continue
 		}
 
+		downloader := s3manager.NewDownloader(sess, func(d *s3manager.Downloader) {
+			d.PartSize = 100 * 1024 * 1024
```

Comment:
nit: probably a comment here stating the units. I am guessing its 100 MB

Reply:
Done. Added the comment